### PR TITLE
Provide ENABLE_STARTUP_FALLBACK property to enable/disable CPU as acc…

### PR DESCRIPTION
**Pull request by WeldonWangwang**
### Details:
 - Provide the option to disable CPU as acceleration (or helper device) at the beginning, but still keep CPU fallback option

### Tickets:
 - *92996*

### CODE REVIEW

File: src/plugins/auto/plugin.cpp [hunk 1]
```diff
@@ -446,6 +448,9 @@ IExecutableNetworkInternal::Ptr MultiDeviceInferencePlugin::LoadNetworkImpl(cons
         auto tmpiter = fullConfig.find(ov::intel_auto::device_bind_buffer.name());
         if (tmpiter != fullConfig.end() && tmpiter->second == PluginConfigParams::YES)
             autoSContext->_bindBuffer = true;
+        auto tmpiter_enableStartupFallback = fullConfig.find(ov::intel_auto::enable_startup_fallback.name());
+        if (tmpiter_enableStartupFallback != fullConfig.end() && tmpiter_enableStartupFallback->second == PluginConfigParams::YES)
+            autoSContext->_startupfallback = false;
```
[the remark this answers is not in the document]
Comment (WeldonWangwang) [reply src/plugins/auto/plugin.cpp:453]: Nice catch, fix it, thank you.

File: src/plugins/auto/auto_schedule.cpp [hunk 2]
```diff
@@ -143,11 +143,13 @@ void AutoSchedule::init(const ScheduleContext::Ptr& sContext) {
                                                                            _loadContext[ACTUALDEVICE].networkPrecision,
                                                                            _autoSContext->_modelPriority);
     }
+    if (!_autoSContext->_startupfallback)
+        _loadContext[ACTUALDEVICE].deviceInfo.config[ov::intel_auto::enable_startup_fallback.name()] = InferenceEngine::PluginConfigParams::NO;
```
Comment (songbell) [inline src/plugins/auto/auto_schedule.cpp:147]: looks like we don't need to carry on this config? do we?

File: src/plugins/intel_cpu/tests/functional/shared_tests_instances/behavior/ov_plugin/properties_tests.cpp [hunk 3]
```diff
@@ -28,7 +28,9 @@ const std::vector<ov::AnyMap> multi_Auto_properties = {
         {ov::device::priorities(CommonTestUtils::DEVICE_CPU), ov::hint::performance_mode(ov::hint::PerformanceMode::LATENCY)},
         {ov::device::priorities(CommonTestUtils::DEVICE_CPU), ov::hint::performance_mode(ov::hint::PerformanceMode::CUMULATIVE_THROUGHPUT)},
         {ov::device::priorities(CommonTestUtils::DEVICE_CPU), ov::intel_auto::device_bind_buffer("YES")},
-        {ov::device::priorities(CommonTestUtils::DEVICE_CPU), ov::intel_auto::device_bind_buffer("NO")}
+        {ov::device::priorities(CommonTestUtils::DEVICE_CPU), ov::intel_auto::device_bind_buffer("NO")},
```
Comment (songbell) [inline src/plugins/intel_cpu/tests/functional/shared_tests_instances/behavior/ov_plugin/properties_tests.cpp:31]: we need some functional tests like mock, to ensure that cpu helper is loaded only when enable_startup_fallback flag enabled

**Review by songbell — COMMENTED** [review]
tests needed

**Review by wangleis — APPROVED** [review]

**Comment by wangleis** [comment]
Need another PR for 22.3 branch

File: src/plugins/auto/auto_schedule.cpp (hunk 2 above)
Comment (WeldonWangwang) [reply src/plugins/auto/auto_schedule.cpp]: Yes, removed it.

File: src/plugins/intel_cpu/tests/functional/shared_tests_instances/behavior/ov_plugin/properties_tests.cpp (hunk 3 above)
Comment (WeldonWangwang) [reply src/plugins/intel_cpu/tests/functional/shared_tests_instances/behavior/ov_plugin/properties_tests.cpp:31]: Done

**Comment by peterchen-intel** [comment]
@WeldonWangwang Please check this failure
build ie ERROR: /home/jenkins/agent/workspace/private-ci/ie/build-linux-ubuntu20/b/repos/openvino/src/tests/unit/auto/auto_startup_fallback_properties_test.cpp:19:10: fatal error: cpp/ie_plugin.hpp: No such file or directory

**Comment by WeldonWangwang** [comment]
> auto_startup_fallback_properties_test.cpp

**Comment by WeldonWangwang** [comment]
> @WeldonWangwang Please check this failure build ie ERROR: /home/jenkins/agent/workspace/private-ci/ie/build-linux-ubuntu20/b/repos/openvino/src/tests/unit/auto/auto_startup_fallback_properties_test.cpp:19:10: fatal error: cpp/ie_plugin.hpp: No such file or directory

Okay, fix it(based on https://github.com/openvinotoolkit/openvino/pull/15229)

File: src/plugins/auto/auto_schedule.cpp [hunk 4]
```diff
@@ -160,7 +160,7 @@ void AutoSchedule::init(const ScheduleContext::Ptr& sContext) {
     bool isActualDevCPU =
         _loadContext[ACTUALDEVICE].deviceInfo.deviceName.find("CPU") !=std::string::npos && !isCumulative;
     // if Actual device is CPU or perf_hint is cumulative, disabled _loadContext[CPU], only use _loadContext[ACTUALDEVICE]
-    if (isActualDevCPU || isCumulative) {
+    if (isActualDevCPU || isCumulative || !_autoSContext->_startupfallback) {
```
Comment (wangleis) [inline src/plugins/auto/auto_schedule.cpp:163]: Where is the logic to catch inference failure on accelerator and check if CPU is candidate device?

File: src/tests/unit/auto/auto_startup_fallback_properties_test.cpp [hunk 5]
```diff
@@ -0,0 +1,157 @@
+// Copyright (C) 2018-2022 Intel Corporation
```
Comment (riverlijunjie) [inline src/tests/unit/auto/auto_startup_fallback_properties_test.cpp:1]:
```suggestion
// Copyright (C) 2018-2023 Intel Corporation
```

File: src/tests/unit/auto/auto_startup_fallback_properties_test.cpp [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+    ON_CALL(*plugin, SelectDevice(_, _, _)).WillByDefault(Return(metaDevices[1]));
+
+    EXPECT_CALL(
+        *core,
+        LoadNetwork(::testing::Matcher<const InferenceEngine::CNNNetwork&>(_),
+                    ::testing::Matcher<const std::string&>(CommonTestUtils::DEVICE_GPU),
+                    ::testing::Matcher<const std::map<std::string, std::string>&>(_)))
+        .Times(1);
+    if (startup_fallback) {
+        std::map<std::string, std::string> test_map = {{"PERFORMANCE_HINT", "LATENCY"}};
```
Comment (riverlijunjie) [inline src/tests/unit/auto/auto_startup_fallback_properties_test.cpp:139]: Can you explain why set "LATENCY" mode here.

**Review by riverlijunjie — APPROVED** [review]
LGTM

File: src/tests/unit/auto/auto_startup_fallback_properties_test.cpp (hunk 6 above)
Comment (WeldonWangwang) [reply src/tests/unit/auto/auto_startup_fallback_properties_test.cpp:139]: When the CPU as an accelerator, PERFORMANCE_HINT must be LATENCY.

File: src/tests/unit/auto/auto_startup_fallback_properties_test.cpp [hunk 7]
```diff
@@ -0,0 +1,157 @@
...
+using ::testing::StrEq;
+using ::testing::Return;
+using ::testing::Property;
+using ::testing::Eq;
+using ::testing::ReturnRef;
+using ::testing::AtLeast;
+using ::testing::AnyNumber;
+using ::testing::InvokeWithoutArgs;
+using ::testing::NiceMock;
+using Config = std::map<std::string, std::string>;
```
Comment (songbell) [inline src/tests/unit/auto/auto_startup_fallback_properties_test.cpp:36]: please check all the headers and usings, simplify if not needed

**Review by wgzintel — APPROVED** [review]

File: src/tests/unit/auto/auto_startup_fallback_properties_test.cpp (hunk 7 above)
Comment (WeldonWangwang) [reply src/tests/unit/auto/auto_startup_fallback_properties_test.cpp:26]: Updated, please review

**Review by wangleis — APPROVED** [review]

**Pull request merged**